board/flags-config.cmake

```diff
@@ -0,0 +1,65 @@
+# This function provides a set of specific flags for each supported board
+# Depending on the processor type. The objective is to optimize for size.
+# Thus, all of the fllowing flags are chosen carefully to reduce binary 
+# footprints.
+
+# Set generic minimization flags for all platforms.
+# These flags are the same for all cross-compilation cases.
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Os -fdata-sections -ffunction-sections")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fomit-frame-pointer -fno-unwind-tables")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-asynchronous-unwind-tables -fvisibility=hidden")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fshort-enums -finline-small-functions")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -findirect-inlining -fno-common") 
+#-flto -fuse-ld=gold # There is an issue with gold link when compiling on 
+# Ubuntu 16. At that point gcc linker did not integrate the flto support 
+# inside and it was a separate plugin that need to be added. Therefore, 
+# this can be added when mlpack Azure CI moves toward Ubuntu 20.
+
+set(BOARD_NAME "" CACHE STRING "Specify Board name to optimize for.")
+
+# Set specific platforms CMAKE CXX flags.
+if(BOARD_NAME MATCHES "RPI0" OR BOARD_NAME MATCHES "RPI1")
```

Comment:
It might be nice to use `string(TOUPPER ...)` on `BOARD_NAME` before these comparisons.  (I'd suggest doing that into a temporary variable, so that we don't change whatever the user specified as `BOARD_NAME`.)  This way, the user can specify `RPI1` or `rpi1` or even `rPi1` and there will be no problem. :+1: